File: src/main/java/club/maxstats/weave/loader/api/Hook.java
package club.maxstats.weave.loader.api;

import org.objectweb.asm.tree.ClassNode;

public abstract class Hook {
    public final String targetClassName;

    public Hook(String targetClassName) {
        this.targetClassName = targetClassName;
    }

    public abstract void transform(ClassNode node, AssemblerConfig cfg);

    public interface AssemblerConfig {
        void computeFrames();
    }
}

File: src/main/kotlin/club/maxstats/weave/loader/api/Hook.kt
package club.maxstats.weave.loader.api

import org.objectweb.asm.tree.ClassNode

public abstract class Hook(public val targetClassName: String) {
    public abstract fun transform(node: ClassNode, cfg: AssemblerConfig)

    public abstract class AssemblerConfig {
        public abstract fun computeFrames()
    }
}
